sdk/java/pulumi/src/test/java/com/pulumi/deployment/DeploymentInvokeDependsOnTest.java

```diff
@@ -0,0 +1,177 @@
+package com.pulumi.deployment;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.pulumi.core.Output;
+import com.pulumi.core.TypeShape;
+import com.pulumi.core.annotations.CustomType;
+import com.pulumi.core.annotations.CustomType.Setter;
+import com.pulumi.core.annotations.Import;
+import com.pulumi.core.annotations.ResourceType;
+import com.pulumi.core.internal.Internal;
+import com.pulumi.deployment.internal.Runner;
+import com.pulumi.resources.Resource;
+import com.pulumi.resources.InvokeArgs;
+import com.pulumi.resources.CustomResource;
+import com.pulumi.resources.CustomResourceOptions;
+import com.pulumi.resources.ResourceArgs;
+import com.pulumi.test.Mocks;
+import com.pulumi.test.TestOptions;
+import com.pulumi.test.internal.PulumiTestInternal;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.Objects;
+import java.util.ArrayList;
+import java.util.concurrent.CompletableFuture;
+import java.util.Optional;
+import java.lang.Thread;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DeploymentInvokeDependsOnTest {
+
+    @AfterEach
+    void cleanup() {
+        PulumiTestInternal.cleanup();
+    }
+
+    @Test
+    void testInvokesDependsOn() {
```

Comment:
We have a unit test here, but the language conformance test `l2-invoke-options-depends-on` is not enabled yet. I spent some time trying to fix program gen to generate something usable, but haven't completed that work yet. The way we generate lambdas is broken, and requires work.